src/Validation/Validator.php

```diff
@@ -1941,6 +1929,51 @@ public function regex($field, $regex, $message = null, $when = null)
         ]);
     }
 
+    /**
+     * Gets the required message for a field
+     *
+     * @param string $field Field name
+     * @return string
+     */
+    public function getRequiredMessage($field)
+    {
+        $defaultMessage = 'This field is required';
+        if ($this->_useI18n) {
+            $defaultMessage = __d('cake', 'This field is required');
+        }
+
+        return isset($this->_presenceMessages[$field])
+            ? $this->_presenceMessages[$field]
+            : $defaultMessage;
+    }
+
+    /**
+     * Gets the notEmpty message for a field
+     *
+     * @param string $field Field name
+     * @return string
+     */
+    public function getNotEmptyMessage($field)
+    {
+        $defaultMessage = 'This field cannot be left empty';
+        if ($this->_useI18n) {
+            $defaultMessage = __d('cake', 'This field cannot be left empty');
+        }
+
+        $notBlankMessage = null;
+        if (isset($this->_fields[$field])) {
```

Comment:
How about doing this check at the beginning of method and simply returning `null` if no validation set is set for this field?  Also makes if wonder if this needs to be a public problem. If it were protected check wouldn't even be needed since `$field` is got by iterating over `$this>_fields` in `errors()`.

Reply:
Seem odd to me to return a validation message even when no validations are set for field provided as argument.